commons/src/main/java/com/powsybl/openrao/commons/NumberRounding.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.powsybl.openrao.commons;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+/**
+ * @author Thomas Bouquet {@literal <thomas.bouquet at rte-france.com>}
+ */
+public final class NumberRounding {
+    private static final double ROUNDING_EPSILON = 1e-8; // "noise" required to avoid side effects when rounding negative powers of 10
+
+    private NumberRounding() {
+    }
+
+    public static int computeNumberOfRelevantDecimals(double value, int defaultDecimals) {
+        return value == (int) value ? defaultDecimals : Math.max(defaultDecimals, (int) -Math.log10(Math.abs(value - (int) value) + ROUNDING_EPSILON) + 1);
+    }
+
+    public static int computeRelevantMarginDecimals(double margin, int defaultDecimals) {
+        if (margin >= 0) {
+            return defaultDecimals; // no violation so no need to be more specific in terms of decimals
+        }
+        // otherwise, if the violation is greater than 1 in absolute value, it is sufficiently high to be seen even if the value is rounded
+        return margin > -1 ? computeNumberOfRelevantDecimals(margin, defaultDecimals) : defaultDecimals;
```

Comment:
i really don't like having functions in commons that only work for a very specific situation (margin between -1 and 0) with generic names like computeNumber of Relevant decimals, and with a "complicated" formula

Reply:
i added docstring and renamed the class to make it more understandable

Reply:
however I don't see any other place in the repo to put this code since it must be mutualized to JSON RaoResult and CNE exporter
maybe when #1158 is merged we can add a ExporterUtils class which will be common to all RaoResult exporters and put this method there, but for the moment let's leave it in commons